components/editor/publish-button.tsx

```diff
@@ -59,6 +62,13 @@ const PublishButton = ({
     log.debug(`Published Adventure: ${adventureId}`);
   };
 
+  const undoChanges = async () => {
+    setIsUndoLoading(true);
+    await resetDevAgent(adventureId as string);
+    window.location.reload();
```

Comment:
path invalidation isn't working properly here for some reason.. so for now a hard reload will do the trick